apps/designer/app/designer/features/sidebar-left/panels/asset-manager/utils.ts

```diff
@@ -0,0 +1,22 @@
+export const getHumanReadableFileSize = (size: number) => {
+  const i = Math.floor(Math.log(size) / Math.log(1024));
+  const types = ["B", "kB", "MB", "GB", "TB"];
+  return `${(size / Math.pow(1024, i)).toFixed(2)} ${types[i]}`;
+};
+
+export const getStartAndEndOfString = (str: string) => {
+  if (str.length > 20) {
```

Comment:
20 is arbitrary number, this won't work depending on the size of the container or font. Usually this is done using CSS, am I right to assume you wanted the ... in the middle of the string? There is a better solution to this that doesn't use arbitrary string length or otherwise detects it

Reply:
One thing I can think of is to create a component that would split a string in two parts, and make the left part truncate using CSS ... this is just what comes into mind right away, maybe there is an even better way

Reply:
I think aiming to truncate before an extension specifically isn't great because if the file doesn't have an extension for some reason, you end up truncating at the end ... macos seems to be truncating in the middle

Reply:
quick googling gave an interesting css truncation appraoch https://www.codeply.com/go/UBmrMAiRsG/css-truncate-ellipsis-in-middle-of-text

Reply:
https://stackoverflow.com/questions/27746218/css-crop-string-in-the-middle

Reply:
done!